### ISSUE TITLE
blacklisted name span includes `mut`

### ISSUE THREAD

**Issue by oli-obk** [Manishearth/rust-clippy#1647]
```rust
let mut foo: i32;
```

lints over the whole span `mut foo`

```
warning: use of a blacklisted/placeholder name `foo`
 --> src/main.rs:2:9
  |
2 |     let mut foo: i32;
  |         ^^^^^^^
  |
  = note: #[warn(blacklisted_name)] on by default
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#blacklisted_name
```

it should just hilight the `foo` part.
